style/js/shield_defs.js

```diff
@@ -798,6 +798,100 @@ export function loadShields(shieldImages) {
     },
   };
 
+  shields["JP:E"] = roundedRectShield("#006747", "white", "white", 2, 1);
+  [
+    "首都高速道路", // Shuto
+    "名古屋高速道路", // Nagoya
+  ].forEach((network) => {
+    shields[network] = {
+      backgroundImage: shieldImages.shield40_jp_urban_expressway,
+      textLayoutConstraint: ShieldText.southHalfellipseTextConstraint,
+      textColor: "white",
+      padding: {
+        left: 3,
+        right: 3,
+        top: 2,
+        bottom: 6,
+      },
+    };
+  });
+  shields["JP:national"] = {
+    backgroundImage: [
+      shieldImages.shield40_jp_national_2,
+      shieldImages.shield40_jp_national_3,
+    ],
+    textLayoutConstraint: ShieldText.ellipseTextConstraint,
+    textColor: "white",
+    padding: {
+      left: 2,
+      right: 2,
+      top: 1,
+      bottom: 5,
+    },
+  };
+  shields["JP:prefectural"] = {
+    backgroundImage: shieldImages.shield40_jp_prefectural,
+    textLayoutConstraint: ShieldText.ellipseTextConstraint,
+    textColor: "white",
+    padding: {
+      left: 3,
+      right: 3,
+      top: 2,
+      bottom: 2,
+    },
+  };
+  [
+    "aichi",
+    "akita",
+    "aomori",
+    "chiba",
+    "ehime",
+    "fukui",
+    "fukuoka",
+    "fukushima",
+    "gifu",
+    "gunma",
+    "hiroshima",
+    "hokkaido",
+    "hyogo",
+    "ibaraki",
+    "ishikawa",
+    "iwate",
+    "kagawa",
+    "kagoshima",
+    "kanagawa",
+    "kochi",
+    "kumamoto",
+    "kyoto",
+    "mie",
+    "miyagi",
+    "miyazaki",
+    "nagano",
+    "nagasaki",
+    "nara",
+    "niigata",
+    "oita",
+    "okayama",
+    "okinawa",
+    "osaka",
+    "saga",
+    "saitama",
+    "shiga",
+    "shimane",
+    "shizuoka",
+    "tochigi",
+    "tokushima",
+    "tokyo",
+    "tottori",
+    "toyama",
+    "wakayama",
+    "yamagata",
+    "yamaguchi",
+    "yamanashi",
+  ].forEach((prefecture) => {
+    shields[`JP:prefectural:${prefecture}`] = shields["JP:prefectural"];
+  });
```

Comment:
There are two tagging schemes for prefectural highways: [`network=JP:prefectural`](https://wiki.openstreetmap.org/wiki/JA%3ATag%3Anetwork%3DJP%3Aprefectural) and [`network=JP:prefectural:*`](https://wiki.openstreetmap.org/wiki/Tag%3Anetwork%3DJP%3Aprefectural%3Anagano), qualified by the prefecture name. `JP:prefectural` is the more common scheme, but both are on the rise.

As far as I can tell, the situation on the ground is similar to county routes in some U.S. states. The shields are uniform nationwide, except for a legend at the bottom that differs by prefecture. However, each prefecture numbers its highways independently, so numbers frequently get reused across borders.